client/js/components/StatsLeaderboard.vue

```diff
@@ -0,0 +1,359 @@
+<template>
+  <div>
+    <div id="stats-table-upper-surface" class="d-flex align-end">
+      <v-select
+        v-model="selectedMetric"
+        :items="metricChoices"
+        label="Select Metric"
+        class="filter-select mr-4 fit"
+        data-cy="metric-select"
+      />
+      <v-select
+        v-model="selectedWeeks"
+        :items="weeks"
+        label="Select Weeks"
+        data-cy="week-select"
+        class="flex-grow-1"
+        menu-props="data-week-select-menu"
+        multiple
+        chips
+        deletable-chips
+      />
+    </div>
+    <v-data-table
+      :items="tableRows"
+      :headers="tableColumns"
+      :sort-by.sync="sortBy"
+      :loading="loading"
+      :item-class="tableRowClass"
+    >
+      <template #[`item.rank`]="{item, value}">
+        <span :data-rank="item.username">{{ value }} </span>
+      </template>
+      <template v-for="week in ['total', ...selectedWeeks]" #[`item.week_${week}`]="{item, value}">
+        <v-tooltip :key="`${item.username}_week_${week}_wins`" top v-if="value">
+          <template #activator="{on, attrs}">
+            <v-chip
+              :key="`${item.username}_week_${week}`"
+              :color="colorForScoreByWeek(week, value)"
```

Comment:
It's a little hard for me to follow, but I feel like this doesn't need to be a method and could instead be a straight lookup in a mapping for performance since there is a finite number of weeks. That feels really nitpicky without any sort of real example to back it up. I guess my major question would be how dynamic is this? The scores seem pretty variable but if you have the data up front you might be able to map them pretty easy.

Reply:
For any given cell the value is not dynamic. A big map would be reasonable but actually I'm thinking it'd be cleaner to simply make a component so that this value can be a computed prop inside it.

Reply:
Pulled this into a component which breaks up the code and allows this to be a computed prop scoped to the one cell rather than a function call